packages/puppeteer-core/src/common/util.ts

```diff
@@ -510,7 +510,7 @@ export async function waitWithTimeout<T>(
     timeout,
   });
 
-  return await Deferred.race([promise, deferred.valueOrThrow()]).finally(() => {
+  return await Deferred.race([promise, deferred]).finally(() => {
```

Comment:
This was causing timeouts in test for BiDi, and they happened on a hook, stopped like 4 test from running.
Created an EsLint rule to prevent this for the feature.